src/IO/SwapHelper.h

```diff
@@ -1,16 +1,19 @@
 #pragma once
-#include <IO/BufferBase.h>
+
+#include <IO/WriteBuffer.h>
 
 namespace DB
 {
-    class SwapHelper
-    {
-    public:
-        SwapHelper(BufferBase & b1_, BufferBase & b2_) : b1(b1_), b2(b2_) { b1.swap(b2); }
-        ~SwapHelper() { b1.swap(b2); }
 
-    private:
-        BufferBase & b1;
-        BufferBase & b2;
-    };
+class SwapHelper
```

Comment:
This class looks strange.

Reply:
I also do not like that, but what exactly could be better?